src/runtime_src/core/include/xrt/xrt_bo.h

```diff
@@ -111,18 +111,22 @@ class bo
    *  Create a BO for peer-to-peer use
    * @var svm
    *  Create a BO for SVM (supported on specific platforms only)
-   *
+   * @var device_memory
+   *  Create a BO from device memory, which is a separate pool of
+   *  memory that is supported on specific platforms only.
+   * 
    * The flags used by xrt::bo are compatible with XCL style
    * flags as define in ``xrt_mem.h``
    */
   enum class flags : uint32_t
   {
-    normal      = 0,
-    cacheable   = XRT_BO_FLAGS_CACHEABLE,
-    device_only = XRT_BO_FLAGS_DEV_ONLY,
-    host_only   = XRT_BO_FLAGS_HOST_ONLY,
-    p2p         = XRT_BO_FLAGS_P2P,
-    svm         = XRT_BO_FLAGS_SVM,
+    normal        = 0,
+    cacheable     = XRT_BO_FLAGS_CACHEABLE,
+    device_only   = XRT_BO_FLAGS_DEV_ONLY,
+    host_only     = XRT_BO_FLAGS_HOST_ONLY,
+    p2p           = XRT_BO_FLAGS_P2P,
+    svm           = XRT_BO_FLAGS_SVM,
+    device_memory = XRT_BO_FLAGS_CARVEOUT,
```

Comment:
device_memory is misleading. Can we call it what it really is -- carveout?

Reply:
@sonals carveout is even worse., please don't use that.